src/Sulu/Bundle/AdminBundle/Resources/js/views/FormOverlayList/tests/FormOverlayList.test.js

```diff
@@ -0,0 +1,276 @@
+/* eslint-disable flowtype/require-valid-file-annotation */
+import mockReact from 'react';
+import {mount} from 'enzyme';
+import {observable} from 'mobx';
+import FormOverlayList from '../FormOverlayList';
+import List from '../../List';
+import Overlay from '../../../components/Overlay';
+import ResourceStore from '../../../stores/ResourceStore';
+import ResourceFormStore from '../../../containers/Form/stores/ResourceFormStore';
+import Form from '../../../containers/Form';
+import ErrorSnackbar from '../ErrorSnackbar';
+
+const React = mockReact;
+
+jest.mock('../../List', () => class ListMock extends mockReact.Component {
+    listStore = {};
+
+    render() {
+        return <div>list view mock</div>;
+    }
+});
+
+jest.mock('../../../containers/Form', () => class ListMock extends mockReact.Component {
+    render() {
+        return <div>form container mock</div>;
+    }
+});
+
+jest.mock('../../../utils/Translator', () => ({
+    translate: jest.fn((key) => key),
+}));
+
+jest.mock('../../../stores/ResourceStore', () => jest.fn(
+    (resourceKey, itemId) => {
+        return {
+            id: itemId,
+        };
+    }
+));
+jest.mock('../../../containers/Form/stores/ResourceFormStore', () => jest.fn(
+    (resourceStore) => {
+        return {
+            id: resourceStore.id,
+        };
+    }
+));
+
+let router;
+beforeEach(() => {
+    router = {
+        attributes: {
+            id: 'test-id',
+            category: 'category-id',
+        },
+        route: {
+            options: {
+                adapters: ['table'],
+                addRoute: 'addRoute',
+                listKey: 'test-list-key',
+                formKey: 'test-form-key',
+                addOverlayTitle: 'app.add_overlay_title',
+                editOverlayTitle: 'app.edit_overlay_title',
+                resourceKey: 'test-resource-key',
+                toolbarActions: ['sulu_admin.add'],
+                routerAttributesToListStore: {0: 'category', 'parentId': 'id'},
+                routerAttributesToFormStore: {0: 'category', 'parentId': 'id'},
+            },
+        },
+    };
+});
+
+test('View should render with closed overlay', () => {
+    const formOverlayList = mount(<FormOverlayList router={router} />);
+
+    expect(formOverlayList.render()).toMatchSnapshot();
+});
+
+test('View should render with opened overlay', () => {
+    const formOverlayList = mount(<FormOverlayList router={router} />);
+    formOverlayList.instance().createFormOverlay(undefined, 'test-list-key');
```

Comment:
I think you should open the overlay using some click event. This way it is not said that it really works...

Reply:
the thing here is that there is no element mounted that we could click. the `FormOverlayList` view does not render any clickable element itself but only sets the callbacks to the embedded `List` view.

This embedded `List` view, which renders the edit-button and sets the add-button to the toolbar, is mocked (see line 17) because we do not want to mock all dependencies of the `List` view in this test-suite in my opinion.